### PR TITLE
Add nand2tetris recipes

### PR DESCRIPTION
**Pull request by CestDiego**
I had to modularized my packages, so I had to make some changes to the
recipes.c

### CODE REVIEW

File: recipes/nand2tetris [hunk 1]
```diff
@@ -1,3 +1,5 @@
 (nand2tetris :fetcher github
              :repo "CestDiego/nand2tetris.el"
-             :files (:defaults "snippets"))
+             :files ("nand2tetris.el"
+                     ("snippets"
+                      ("nand2tetris-mode" "snippets/nand2tetris-mode/*"))))
```
Comment (purcell) [inline recipes/nand2tetris]: Just use `"snippets"` here to coy the whole subdir recursively.

File: recipes/nand2tetris-core [hunk 2]
```diff
@@ -0,0 +1,3 @@
+(nand2tetris-core :fetcher github
+                  :repo "CestDiego/nand2tetris.el"
+                  :files ("nand2tetris-core.el"))
```
Comment (purcell) [inline recipes/nand2tetris-core]: Would anybody use `nand2tetris-core` without `nand2tetris` or `nand2tetris-assembler`? This packaging feels too fine-grained to me, with *perhaps* the exception of `company-nand2tetris`.

Comment (CestDiego) [inline recipes/nand2tetris-core]: no, but how else do I  require the package? should I include core with nand2tetris...and then in company-nand2tetris I should require nand2tetris? (which requires core)

Comment (purcell) [inline recipes/nand2tetris-core]: You don't need package requirements between files in the same package. So put all the related files in one package, and then they can just `require` each other. The package metadata would go in the "main" `.el` file - the one with the name matching the package name. Hope that helps.

Comment (CestDiego) [inline recipes/nand2tetris-core]: I modified it so the core is included in the company then :D

**Comment by purcell** [comment]
In company-nand2tetris.el, this is invalid because you need to use proper version numbers:
```
;; Package-Requires: ((names "Lookup.Latest.Version") (nand2tetris "0.0.1") (company "0.5") (cl-lib "0.5.0"))
```

Same in nand2tetris.el:
```
;; Package-Requires: ((names "Lookup.Latest.Version") (yasnippet "0.8.0"))
```

and nand2tetris-assembler.el:
```
;; Package-Requires: ((names "Lookup.Latest.Version") (nand2tetris "0.0.1"))
```

Don't depend on `yasnippet` because it's not core to your package, and not everybody wants to have it installed. Instead, just soft-require it: `(eval-when-compile (require 'yasnippet nil t))`.

**Comment by purcell** [comment]
You can test all these recipes locally using `C-c C-c` -- this will let you check they all install cleanly. :-)

**Comment by milkypostman** [comment]
ping @CestDiego

**Comment by CestDiego** [comment]
@purcell @milkypostman I'm sorry for my absence, well it turns out that I thought "Lookup.Latest.Version" was actually not a function that Melpa had that would make it search for latest version of the package and then install it :P

**Comment by purcell** [comment]
:+1:

**Comment by CestDiego** [comment]
Thank you for your help :) I will take into consideration compilation errors in the next packages :D

**Pull request merged**